### DISCUSSION TITLE
request help: does APISIX support gray release after integration with Nacos

### DISCUSSION THREAD

**Discussion by zzywq** [apache/apisix#6172] [Q&A]
### Issue description

集成注册中心后（nacos），能否支持灰度发布 
例如 客户端能够通过一个版本号或者参数，能够精准的路由到满足条件和后端服务节点上去。

### Environment

集成注册中心后（nacos），能否支持灰度发布

### Steps to reproduce

集成注册中心后（nacos），能否支持灰度发布

### Actual result

集成注册中心后（nacos），能否支持灰度发布

### Error log

集成注册中心后（nacos），能否支持灰度发布

### Expected result

集成注册中心后（nacos），能否支持灰度发布

**Comment by zzywq** [comment]
其实提这个需求，注意的应用场景在于本地开发环境，能够更加header  信息精准的绑定到指定的后端节点上面去，
想到的方案是能够自定义负载均衡策略，或者nacos 服务中的路由元数据就行过滤

**Comment by tzssangglass** [comment]
This is not an issue, I will move to discussion.

**Reply by tzssangglass** [reply]
It seems to be a good problem, we have the grey release plugin, but this plugin is not easy to integrate with the service discovery, the main conflict is that the upstream nodes information is not in APISIX.

**Reply by zzywq** [reply]
Can the load balancing strategy be customized? If so, you only need to obtain the parameters in the request header. As long as apisix obtains the Nacos routing metadata in the service discovery API, retains it, and then matches the parameters in the request header, this function can be realized.

**Reply by tzssangglass** [reply]
Yes, you can achieve it yourself
